extendr-api/src/wrapper/externalptr.rs

```diff
@@ -82,24 +82,25 @@ impl<T: Any + Debug> ExternalPtr<T> {
     /// An ExternalPtr behaves like a Box except that the information is
     /// tracked by a R object.
     pub fn new(val: T) -> Self {
+        use std::ffi::c_void;
         unsafe {
             // This allocates some memory for our object and moves the object into it.
-            let boxed = Box::new(val);
+            let v: Box<dyn Any> = Box::new(val);
 
             // This constructs an external pointer to our boxed data.
             // into_raw() converts the box to a malloced pointer.
-            let robj = Robj::make_external_ptr(Box::into_raw(boxed), r!(()));
```

Comment:
Does `Robj::make_external_ptr()` is used anywhere else? can we resort to calling `extendr_api` methods rather than digging into `libR-sys`?

Reply:
Also, here I see double indirection, which I am not sure is necessary or efficient.

Reply:
When I submitted this PR, the tests passed locally. Now I'm changing it because something very odd is happening.

I don't want to use `Robj::make_external_ptr`, because it wants to protect `tag` and `prot`, but
those are protected by being part of an `ExternalPtr`.

The indirection is what gives me the ability to pick up the type information from the pointer.
As (apparently) the typeid is stored in a `Box<dyn Any>` rather than a `*mut T`. 
I don't know where this is written :(